src/main/java/org/openstreetmap/atlas/checks/validation/tag/ConditionalRestrictionCheck.java

```diff
@@ -0,0 +1,223 @@
+        }
+        return Optional.empty();
+    }
+
+    @Override
+    protected List<String> getFallbackInstructions()
+    {
+        return FALLBACK_INSTRUCTIONS;
+    }
+
+    private boolean containsTransportationMode(final String key)
+    {
+        final String[] parts = key.split(":");
+        for (final String part : parts)
+        {
+            if (isTransportationMode(part))
+            {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    private boolean isAccessType(final String value)
+    {
+        return "access".equals(value);
+    }
+
+    private boolean isAccessValue(final String value)
+    {
+        return ACCESS_RESTRICTION_VALUE.contains(value);
+    }
+
+    private boolean isDirection(final String value)
+    {
+        return DIRECTION.contains(value);
+    }
+
+    private boolean isKeyValid(final String key)
+    {
+        final String[] parts = key.split(":");
+        // access:lanes is a valid exception for lanes on second position
+        final boolean isAccessLanes = parts.length > 2
+                && (isAccessType(parts[0]) && isLanes(parts[1]));
+        switch (parts.length)
+        {
+            // starts with 2 because they all contain the conditional part
+            case TWO_PARTS:
+                return isRestrictionType(parts[0]) || isTransportationMode(parts[0]);
+            case THREE_PARTS:
+                final boolean isRestrictionTypeFormat = isRestrictionType(parts[0])
+                        && (isTransportationMode(parts[1]) || isDirection(parts[1]));
+                final boolean isTransportTypeFormat = isTransportationMode(parts[0])
+                        && (isDirection(parts[1]) || isLanes(parts[1]));
+                return isRestrictionTypeFormat || isTransportTypeFormat || isAccessLanes;
+            case FOUR_PARTS:
+                final boolean isRestrictionTransport = isRestrictionType(parts[0])
+                        && isTransportationMode(parts[1]);
+                final boolean isTransportLanes = isTransportationMode(parts[0])
+                        && isLanes(parts[1]);
+                return (isRestrictionTransport || isTransportLanes || isAccessLanes)
+                        && isDirection(parts[2]);
+            default:
+                return false;
+        }
+    }
+
+    private boolean isLanes(final String value)
+    {
+        return "lanes".equals(value);
+    }
+
+    private boolean isNotLanesType(final String key)
+    {
+        final String[] parts = key.split(":");
+        return !isLanes(parts[0]);
+    }
+
+    private boolean isRestrictionType(final String value)
+    {
+        return RESTRICTION_TYPES.contains(value);
+    }
+
+    private boolean isTransportationMode(final String value)
+    {
+        return TRANSPORTATION_MODE.contains(value);
+    }
+
+    private boolean isValueValid(final String value, final String key)
+    {
+        boolean isValid = true;
```

Comment:
I think this variable is not required. We can simply return true/false based on the condition. For example: ` if (!trimmedPart.isEmpty() && !isAccessValue(trimmedPart))
                        {
                            return false;
                        }`